lazy-pages/src/lib.rs

```diff
@@ -20,68 +20,187 @@
-            lazy_pages_info.insert(LazyPage(p), key);
-        })
-    });
+    pub released_lazy_pages: BTreeMap<PageNumber, Option<PageBuf>>,
+
+    #[deprecated]
+    /// Keys in storage for each lazy page.
+    pub lazy_pages_info: BTreeMap<PageNumber, Vec<u8>>,
+}
+
+thread_local! {
+    // NOTE: here we suppose, that each contract is executed in separate thread.
+    // Or may be in one thread but consequentially.
+
+    /// Identify whether signal handler is set for current thread
+    static LAZY_PAGES_ENABLED: RefCell<bool> = RefCell::new(false);
+    /// Lazy pages context for current execution
+    static LAZY_PAGES_CONTEXT: RefCell<LazyPagesExecutionContext> = RefCell::new(Default::default());
+}
+
+#[deprecated]
+pub fn get_lazy_pages_numbers() -> Vec<PageNumber> {
+    LAZY_PAGES_CONTEXT.with(|ctx| ctx.borrow().lazy_pages_info.keys().copied().collect())
 }
 
-/// Returns vec of not-accessed wasm lazy pages
-pub fn available_pages() -> Vec<LazyPage> {
-    LAZY_PAGES_INFO.with(|lazy_pages_info| lazy_pages_info.borrow().keys().copied().collect())
+/// Returns current wasm mem buffer pointer, if it's set
+pub fn get_wasm_mem_addr() -> Option<HostPointer> {
+    LAZY_PAGES_CONTEXT.with(|ctx| ctx.borrow().wasm_mem_addr)
 }
 
-/// Set current wasm memory begin addr
-pub fn set_wasm_mem_begin_addr(wasm_mem_begin: HostPointer) {
-    WASM_MEM_BEGIN.with(|x| *x.borrow_mut() = wasm_mem_begin);
+/// Returns current wasm mem buffer size, if it's set
+pub fn get_wasm_mem_size() -> Option<usize> {
+    LAZY_PAGES_CONTEXT.with(|ctx| ctx.borrow().wasm_mem_size)
+}
+
+/// Set current wasm memory begin addr in global context
+pub fn set_wasm_mem_begin_addr(wasm_mem_addr: HostPointer) {
+    LAZY_PAGES_CONTEXT.with(|ctx| {
+        let _ = ctx.borrow_mut().wasm_mem_addr.insert(wasm_mem_addr);
+    });
+}
+
+/// Set current wasm memory size in global context
+pub fn set_wasm_mem_size(wasm_mem_size: usize) {
+    LAZY_PAGES_CONTEXT.with(|ctx| {
+        let _ = ctx.borrow_mut().wasm_mem_size.insert(wasm_mem_size);
+    });
 }
 
 /// Reset lazy pages info
-pub fn reset_info() {
-    LAZY_PAGES_INFO.with(|x| x.replace(BTreeMap::new()));
-    RELEASED_LAZY_PAGES.with(|x| x.borrow_mut().clear());
-    WASM_MEM_BEGIN.with(|x| *x.borrow_mut() = 0);
+pub fn reset_context() {
+    LAZY_PAGES_CONTEXT.with(|ctx| *ctx.borrow_mut() = Default::default());
 }
 
 /// Returns vec of lazy pages which has been accessed
-pub fn released_pages() -> Vec<LazyPage> {
-    RELEASED_LAZY_PAGES.with(|x| x.borrow().keys().copied().collect())
+pub fn get_released_pages() -> Vec<PageNumber> {
+    LAZY_PAGES_CONTEXT.with(|ctx| ctx.borrow().released_lazy_pages.keys().copied().collect())
 }
 
 /// Returns whether lazy pages env is enabled
 pub fn is_enabled() -> bool {
     LAZY_PAGES_ENABLED.with(|x| *x.borrow())
 }
 
+#[deprecated]
+/// Set storage `key` for `page` in global context
+pub fn set_lazy_page_info(page: PageNumber, key: &[u8]) {
+    LAZY_PAGES_CONTEXT.with(|ctx| {
```

Comment:
in general, we can just leave here empty fn due to the fact its already deprecated

Reply:
No we cannot because it's used in old runtimes